internal/pkg/agent/application/upgrade/crash_checker_test.go

```diff
@@ -123,7 +125,35 @@ func TestChecker(t *testing.T) {
 		}
 
 		canc()
-		require.Error(t, err)
+		require.ErrorContains(t, err, "service restarted '3' times within '0.1' seconds")
+	})
+
+	t.Run("fails when pid remains 0", func(t *testing.T) {
+		const startingPID = 0
+		pider := &testPider{pid: startingPID}
+		ch, errChan := testableChecker(t, pider)
+		ctx, canc := context.WithCancel(context.Background())
```

Comment:
[Suggestion]
Please, I don't see any value in using a 1/2 word here
```suggestion
		ctx, cancel := context.WithCancel(context.Background())
```